XTS/WSCF/classes/com/arjuna/mw/wscf/protocols/ProtocolManager.java

```diff
@@ -118,14 +119,18 @@ public synchronized final void initialise ()
                 HLSProvider hlsProvider = clazz.getAnnotation(HLSProvider.class);
                 String serviceType = hlsProvider.serviceType();
                 wscfLogger.i18NLogger.info_protocols_ProtocolManager_4(className, serviceType);
-                Object object = clazz.newInstance();
+                Object object = clazz.getDeclaredConstructor().newInstance();
                 _protocols.put(serviceType, object);
             } catch (InstantiationException ie) {
                 wscfLogger.i18NLogger.error_protocols_ProtocolManager_5(className, ie);
             } catch (IllegalAccessException iae) {
                 wscfLogger.i18NLogger.error_protocols_ProtocolManager_5(className, iae);
-            }
-		}
+            } catch (InvocationTargetException e) {
+		wscfLogger.i18NLogger.error_protocols_ProtocolManager_5(className, e);
+	    } catch (NoSuchMethodException e) {
+		wscfLogger.i18NLogger.error_protocols_ProtocolManager_5(className, e);
+	    }
+	}
```

Comment:
I'm not sure you can re-use the same error message here.

Reply:
@jmfinelli any suggestion what message should i use here? I am thinking to use this message "Unable to create new instance of {0}" for `NoSuchMethodException` and for `NoSuchMethodException` i think "Unable to instantiate protocol implementation class {0} caught exception" i.e. `error_protocols_ProtocolManager_5` is correct. WDYT?

Reply:
@jmfinelli any suggestion what message should i use here? I am thinking to use this message "Unable to find constructor for protocol implementation class {0} caught exception" for `NoSuchMethodException` and for `InvocationTargetException` i think "Unable to invoke constructor for protocol implementation class {0} caught exception" . WDYT?

Reply:
Thank you @jmfinelli as  here the message will be same for all 4 Exceptions so think we can add it in one line using `|`:
``` 
catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
  wscfLogger.i18NLogger.error_protocols_ProtocolManager_5(className, ie);
} 
```